### PR TITLE
Fix and UT for testing segment analysis merge

### PR DESCRIPTION
**Pull request by himanshug**
Fixes regression from https://github.com/druid-io/druid/pull/2107

### CODE REVIEW

**Comment by fjy** [comment]
:+1:

File: processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java [hunk 1]
```diff
@@ -149,7 +149,7 @@ public SegmentAnalysis apply(SegmentAnalysis arg1, SegmentAnalysis arg2)
 
             List<Interval> newIntervals = null;
             if (query.hasInterval()) {
-              newIntervals = arg1.getIntervals();
+              newIntervals = new ArrayList<>(arg1.getIntervals());
```
Comment (xvrl) [inline processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java]: Can we add a comment?

Comment (himanshug) [inline processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java]: it is just that the List returned by arg1.getIntervals() is immutable , so it was failing with UnsupportedOperationException. I was probably biased and "thought" that the explanation was trivial so dint add the comment... will add it to make it clear.

Comment (himanshug) [inline processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java]: @xvrl https://github.com/druid-io/druid/pull/2131

**Pull request merged**